src/local-history/local-history-manager.ts

```diff
@@ -75,93 +72,117 @@ export class LocalHistoryManager {
 
-            // Limits the number of entries based on the preference 'local-history.maxEntriesPerFile`.
-            items = items.slice(0, this.localHistoryPreferenceService.maxEntriesPerFile);
-            vscode.window.showQuickPick(items,
-                {
-                    placeHolder: `Please select a local history revision for '${path.basename(textEditor.document.fileName)}'`,
-                    matchOnDescription: true,
-                    matchOnDetail: true
-                }).then((selection) => {
-                    // User made final selection.
-                    if (!selection) {
-                        return;
-                    }
-
-                    // Get the file system path for the selection.
-                    const selectionFsPath = path.join(`${workspaceFolderPath}`, selection.description!);
-
-                    // Show the diff between the active editor and the selected local history file.
-                    this.displayDiff(vscode.Uri.file(selectionFsPath), textEditor.document.uri);
-                });
+                        // Get the file system path for the selection.
+                        const selectionFsPath = path.join(`${hashedFolderPath}`, selection.description!);
+
+                        // Show the diff between the active editor and the selected local history file.
+                        this.displayDiff(vscode.Uri.file(selectionFsPath), textEditor.document.uri);
+                    });
+            });
         }
         else {
             vscode.window.showInformationMessage('File does not belong to a workspace. Please save.');
         }
     }
 
-    /**
-     * Return the timestamp in the following format: 2020-04-06 10:25:50.123.
-     */
-    private getCurrentTime(): string {
-        let time = new Date();
-        time = new Date(time.getTime() - (time.getTimezoneOffset() * 60000));
-        return time.toISOString().substring(0, 23).replace(/[T]/g, ' ');
-    }
-
     /**
      * Save the current context of the active editor.
      */
     public async saveActiveEditorContext(document: vscode.TextDocument): Promise<void> {
         const timestamp = this.getCurrentTime();
         const timestampForFileName = timestamp.replace(/[-:. ]/g, '');
-        const timestampForProperty = timestamp.substring(0, 19);
         const fileFullPath = path.parse(document.fileName);
         const historyFileName = `${fileFullPath.name}_${timestampForFileName.substring(0, 14)}_${timestampForFileName.substring(14, 17)}${fileFullPath.ext}`;
 
         if (vscode.workspace.getWorkspaceFolder(document.uri) !== undefined) {
-            const workspaceFolderPath: string = vscode.workspace.getWorkspaceFolder(document.uri)!.uri.fsPath;
+            const localHistoryFolderPath = path.join(os.homedir(), '.local-history');
+            const hashedEditorPath = checksum(document.fileName);
+            const hashedFolderPath = path.join(localHistoryFolderPath, hashedEditorPath);
```

Comment:
we can change it to  `const hashedFolderPath = path.join(os.homedir(), '.local-history', hashedEditorPath);`